activiti-core/activiti-api-impl/activiti-api-runtime-shared-impl/src/main/java/org/activiti/runtime/api/event/internal/VariableCreatedListenerDelegate.java

```diff
@@ -33,32 +37,47 @@ public class VariableCreatedListenerDelegate implements ActivitiEventListener {
 
     private final VariableEventFilter variableEventFilter;
 
+    private final ProcessExtensionService processExtensionService;
+
+
     public VariableCreatedListenerDelegate(
         List<VariableEventListener<VariableCreatedEvent>> listeners,
         ToVariableCreatedConverter converter,
-        VariableEventFilter variableEventFilter) {
+        VariableEventFilter variableEventFilter, ProcessExtensionService processExtensionService) {
         this.listeners = listeners;
         this.converter = converter;
         this.variableEventFilter = variableEventFilter;
+        this.processExtensionService = processExtensionService;
     }
 
     @Override
     public void onEvent(ActivitiEvent event) {
         if (event instanceof ActivitiVariableEvent) {
             ActivitiVariableEvent internalEvent = (ActivitiVariableEvent) event;
+            Extension extension = processExtensionService.getExtensionsForId(internalEvent.getProcessDefinitionId());
```

Comment:
I think it would be better to inject processExtensionService in the the converter and not here in VariableCreatedListenerDelegate. In this way you can keep a single main method `from` in the converter.